Sources/Fuzzilli/FuzzIL/TypeCollectionStatus.swift

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Unit of code that can be executed, scored, and mutated.
+///
+/// There are a few invariants of programs that are enforced by the engine:
+///
+/// * The empty program is valid
+/// * All input variables must have previously been defined
+/// * Variables have increasing numbers starting at zero and there are no holes
+/// * An instruction always produces a new output variable (SSA form)
+/// * The first input to a Copy operation must be a variable produced by a Phi operation (SSA form)
+///
+/// These invariants can be verified at any time by calling check().
+///
+/// The possible outcome of a program execution.
+public enum TypeCollectionStatus: Equatable {
+    case success
+    case error
+    case timeout
+    case notAttempted
+
+    public init(typeCollectionOutcome: ExecutionOutcome){
```

Comment:
Maybe make it `init(fromExecutionOutcome typeCollectionOutcome: ExecutionOutcome)` or even just `init(from typeCollectionOutcome: ExecutionOutcome)` so it reads more "swifty"?